src/express/outgoing/manager.ts

```diff
@@ -0,0 +1,76 @@
+import { Request } from 'express';
+import * as FormData from 'form-data';
+import { StatusCodes } from 'http-status-codes';
+import mongoose from 'mongoose';
+import { ServerError } from '../error';
+import { handleFileUpload } from '../file';
+import { dropboxService, serviceErrorHandler, statusService, storageService, transfersCrud } from '../services';
+import { checkPhonebookUsers } from '../user/manager';
+import { IFileFlow, INewOutgoingFile, IOutgoingFile, IOutgoingFileFilters } from './interface';
+
+export const createOutgoingFile = async (
+    userId: string,
+    req: Request,
+    newFile: INewOutgoingFile,
+): Promise<IOutgoingFile> => {
+    if (!checkPhonebookUsers(newFile.to))
+        throw new ServerError(StatusCodes.INTERNAL_SERVER_ERROR, 'Failed to get users.');
+
+    const file: IOutgoingFile = (
+        await transfersCrud
+            .post(`/${userId}/outgoing`, { ...newFile, from: userId })
+            .catch(serviceErrorHandler('Failed to create file'))
+    ).data;
+
+    await handleFileUpload(req, async (fileStream) => {
+        const formData = new FormData();
+        formData.append('file', fileStream);
+        return storageService
+            .post(`/bucket/${file.from}/object/${file._id}`, formData, {
+                headers: formData.getHeaders(),
+            })
+            .catch(
+                serviceErrorHandler('Failed to upload file', async () => {
+                    await transfersCrud.delete(`/${userId}/outgoing/${file._id}`);
+                }),
+            );
+    });
+
+    const { _id: fileId, createdAt, ...dropboxBody } = file;
+    await dropboxService.post(`/v1/request`, { id: fileId, fileId, ...dropboxBody }).catch(
+        serviceErrorHandler('Failed to send outgoing file to dropbox', async () => {
+            await transfersCrud.delete(`/${userId}/outgoing/${fileId}`);
+            await storageService.delete(`/bucket/${file.from}/object/${fileId}`);
+        }),
+    );
+
+    return file;
+};
+
+export const getOutgoingFileById = async (userId: string, fileId: mongoose.Types.ObjectId): Promise<IOutgoingFile> => {
+    const response = await transfersCrud
+        .get(`/${userId}/outgoing/${fileId}`)
+        .catch(serviceErrorHandler('Failed to get file'));
+    return response.data;
+};
+
+export const getOutgoingFiles = async (filters: IOutgoingFileFilters): Promise<IOutgoingFile[]> => {
+    const response = await transfersCrud
+        .get(`/${filters.from}/outgoing`, { params: filters })
```

Comment:
make userId function parameter, (changes in controller as well)
```suggestion
        .get(`/${userId}/outgoing`, { params: { ...filters, from: userId } })
```

Reply:
done